script.js

```diff
@@ -1,4 +1,154 @@
+var userWins = 0;
+var computerWins = 0;
+var userName = "";
+var mode = "";
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Check if player enter username
+  if (input != "" && userName == "") {
+    userName = input;
+    return `Your username is ${userName}. Pls key in normal or reversed or korean for game mode.`;
+  }
+  // Normal mode selection
+  if (input == "normal") {
+    mode = input;
+    return `You have selected ${mode} mode.`;
+  } // Reversed mode selection
+  if (input == "reversed") {
+    mode = input;
+    return `You have selected ${mode} mode.`;
+  }
+  // Korean mode selection
+  if (input == "korean") {
+    mode = input;
+    return `You have selected ${mode} mode.`;
+  }
+
+  //Normal mode game play
+  if (mode == "normal") {
+    var getNumber = sortNum();
+    var returnType = ScissorsPaperStone(getNumber);
+    console.log(returnType);
+    var returnInput = input.toLowerCase();
+    var myOutputValue = sps(returnType, returnInput);
+    return myOutputValue;
+  } // Reverse mode game play
+  if (mode == "reversed") {
+    var getNumber = sortNum();
+    var returnType = ScissorsPaperStone(getNumber);
+    var returnInput = input.toLowerCase();
+    var myOutputValue = reversedSps(returnType, returnInput);
+    return myOutputValue;
+  } // Korean mode game play
+  if (mode == "korean") {
+    var getNumber = sortNum();
+    var returnType = ScissorsPaperStone(getNumber);
+    var returnInput = input.toLowerCase();
```

Comment:
Nice accounting for when the user randomly capitalise their inputs 
Like the order of the flow where changes to game modes are evaluated first 
and all playing-of-the-game comes later